internal/db/collection_index.go

```diff
@@ -181,6 +181,22 @@ func (c *collection) deleteIndexedDoc(
 	return nil
 }
 
+func (c *collection) deleteIndexedDocWithID(
+	ctx context.Context,
+	docID client.DocID,
+) error {
+	doc, err := c.get(
```

Comment:
suggestion: It felt a little odd to see the code fetching the doc in order to delete it - I had to chase down the implementation of `deleteIndexedDoc` to realise why - whilst it is kind of obvious once you are in the sec. index mindset, it felt strange briefly stepping into the codespace and it might be worth a comment here to save future people chasing down into the depths of the index code.

Reply:
done